### PR TITLE
RA-988 : Added View / Edit Capability for Encounters

### PR DESCRIPTION
**Pull request by tendomatGitHub**
https://issues.openmrs.org/browse/RA-988

Hello @dkayiwa , @djazayeri  Please checkout these and see if am on the right track...Not yet written a Javascript controller

### CODE REVIEW

File: omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp [hunk 1]
```diff
@@ -37,3 +37,71 @@
     </p>
 </form>
 
+<hr>
+<fieldset class="right" ng-show="encounterType.uuid">
```
Comment (wluyima) [inline omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: I believe the audit info already exists on the form?

File: omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp [hunk 2]
```diff
@@ -37,3 +37,71 @@
...
+        {{ encounterType.auditInfo.changedBy | omrsDisplay }}
+        ${ui.message('general.onDate')}
+        {{ encounterType.auditInfo.dateChanged | serverDate }}
+    </p>
+</fieldset
+
+
+
+<hr>
+<fieldset class="right" ng-show="encounterType.uuid">
```
Comment (wluyima) [inline omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: Why are you adding these sections  for audit info?

**Comment by wluyima** [comment]
The PR seems like it has duplicate code

**Comment by tendomatGitHub** [comment]
@wluyima  thank you very much for the Response,am really new to AngularJS /GSP,and greater still contributing  to RefApp ,, do you have any links to very nice and simple materials the couple  AngularJS /GSP supporting openmrs Refapp..The Documentations i've gone through are very good men of solid experience and a little abstract for starters ,in otherwords they skip the foundations  of how to use AngularJS and GSP to manipulate and produce RefApp.Thanks alot

**Comment by tendomatGitHub** [comment]
I have however been reading several hours trying to back engineer the snippets,,so i really do not understand what and how to use most of those elements in the gsp templates..
tutorialspoint.com has nice tutorials on JSP and Groovy but interestingly ,gsp is missing.

File: omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp (hunk 2 above)
Comment (tendomatGitHub) [reply omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: @wluyima  I was thinking of separating concerns on the same page i.e put the different functionalities in different fieldsets

Comment (tendomatGitHub) [reply omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: However am re-editing the page to remove that,i was thinking of introducing another page,but looks like that will be reinventing the wheel as most of the Encounter attributes have been taken care of @edit.gsp

**Comment by tendomatGitHub** [comment]
@wluyima Please checkout the edit i have done so far and comment.Thanks alot

File: omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp [hunk 3]
```diff
@@ -18,11 +18,20 @@
         ${ui.message('general.onDate')}
         {{ encounterType.auditInfo.dateChanged | serverDate }}
     </p>
+    
+    <p>
```
Comment (wluyima) [inline omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: Why is this for?

File: omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp [hunk 4]
```diff
@@ -18,11 +18,20 @@
...
+        ${ui.message('adminui.location.address')}:
+    </p>
 </fieldset>
 
 
 <form class="simple-form-ui" name="encounterTypeForm" novalidate ng-submit="save()">
     <p>
+        <label>${ui.message('adminui.location.address')}</label>
+        <input ng-model="encounterType.location" required/>
+    </p>
```
Comment (wluyima) [inline omod/src/main/webapp/pages/metadata/encounters/encountertypes/templates/edit.gsp]: Why are you adding this inout, I thought the ticket says adding view/edit privilege fields

**Comment by wluyima** [comment]
@tendomatGitHub did you take a look at https://wiki.openmrs.org/x/7ADn and https://wiki.openmrs.org/x/h5UvAg?

**Comment by dkayiwa** [comment]
@tendomatGitHub did you see @wluyima's comment above?

**Comment by tendomatGitHub** [comment]
@wluyima 
@dkayiwa 

I looked through the documentation.Please look at these templates and comment.Am yet to introduce handler methods in the concerned controller class.Thanks

**Comment by dkayiwa** [comment]
@tendomatGitHub on the JIRA ticket, you need to attach screenshots with the effects of your changes, as advised at https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
@tendomatGitHub did you see the above comments?

**Comment by dkayiwa** [comment]
Ping me to reopen when you resume working on this.

**Comment by tendomatGitHub** [comment]
ok

**Comment by tendomatGitHub** [comment]
@dkayiwa can you re-open this?Iam resuming with it.

**Comment by dkayiwa** [comment]
@tendomatGitHub  are you still interested in finishing this up?

**Comment by tendomatGitHub** [comment]
Yes though i have taken some time off..Just a question do you think it's good praactice to include this feature in OWA instead ?

**Comment by dkayiwa** [comment]
Can you ask this on talk?

**Comment by tendomatGitHub** [comment]
sure did so

**Comment by dkayiwa** [comment]
Closed because of work at https://github.com/openmrs/openmrs-owa-obsadmin as noted on the ticket.

**Comment by tendomatGitHub** [comment]
@dkayiwa  Alright  thanks

**Pull request closed without merging**